embassy/src/fmt.rs

```diff
@@ -195,6 +195,10 @@ macro_rules! unwrap {
     }
 }
 
+#[cfg(feature = "defmt-timestamp-uptime")]
+// defmt offers a disply hint for microseconds so we convert from millis
+defmt::timestamp! {"{=u64:us}", crate::time::Instant::now().as_millis() * 1_000 }
```

Comment:
Could you add `from_micros, as_micros` to `Instant`, and use that instead? It'll be slightly more precise, especially for stm32 which currently defaults to a tick rate of 1mhz. It's also nice for consistency, because `Duration` has it.

Reply:
Done